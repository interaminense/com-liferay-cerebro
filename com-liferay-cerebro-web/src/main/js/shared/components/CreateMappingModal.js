import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';

import Button from './Button';
import Form from './form';
import Modal from './modal';
import sendRequest from 'shared/util/request';

const TYPES = [
	{
		name: Liferay.Language.get('string'),
		value: 'string'
	},
	{
		name: Liferay.Language.get('number'),
		value: 'number'
	},
	{
		name: Liferay.Language.get('boolean'),
		value: 'boolean'
	},
	{
		name: Liferay.Language.get('date'),
		value: 'date'
	}
];

class CreateMappingModal extends Component {
	@autobind
	handleSubmit(formValues) {
		sendRequest({
			data: {
				name: formValues.name,
				type: formValues.type.value
			},
			method: 'PUT',
			path: 'contacts/contacts_mapping'
		}).then(this.props.onClose);
	}

	@autobind
	handleTypeChange(newVal) {
		this.state.selectedType = newVal;
	}

	render() {
		const {props: {onClose}, state: {selectedType}} = this;

		return (
			<Modal elementClasses="create-mapping-modal-root" size="lg">
				<Modal.Header
					onClose={onClose}
					title={Liferay.Language.get('create-new-scv-field')}
				/>

				<Form onSubmit={this.handleSubmit}>
					<Form.Group autoFit inline>
						<Form.Input
							label={Liferay.Language.get('new-field-name')}
							name="name"
							placeholder={Liferay.Language.get(
								'enter-new-field-name'
							)}
						/>

						<div>
							<label>{Liferay.Language.get('field-type')}</label>

							<Form.SearchableSelect
								buttonPlaceholder={
									(selectedType && selectedType.name) ||
									Liferay.Language.get('select')
								}
								caretDouble
								initialValue={selectedType}
								items={TYPES}
								name="type"
								onSelect={this.handleTypeChange}
								selectedItem={selectedType}
								showSearch={false}
							/>
						</div>
					</Form.Group>

					<Modal.Footer>
						<Button onClick={onClose}>
							{Liferay.Language.get('cancel')}
						</Button>

						<Button display="primary" type="submit">
							{Liferay.Language.get('create')}
						</Button>
					</Modal.Footer>
				</Form>
			</Modal>
		);
	}
}

CreateMappingModal.PROPS = {
	onClose: Config.func()
};

CreateMappingModal.STATE = {
	selectedType: Config.object()
};

export default CreateMappingModal;