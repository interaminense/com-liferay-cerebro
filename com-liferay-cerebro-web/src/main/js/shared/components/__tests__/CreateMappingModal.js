import CreateMappingModal from '../CreateMappingModal';
import {renderWithStore} from 'test/mock-store';

describe('CreateMappingModal', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = renderWithStore(CreateMappingModal);

		expect(component).toMatchSnapshot();
	});
});