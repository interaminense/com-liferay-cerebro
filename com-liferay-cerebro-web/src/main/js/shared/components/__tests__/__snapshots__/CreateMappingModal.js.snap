// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMappingModal should render 1`] = `
<div class="modal-dialog modal-lg create-mapping-modal-root">
  <div class="modal-content">
    <div class="modal-header">
      <h4 class="modal-title">create-new-scv-field</h4>
      <button class="button-root btn btn-secondary close" type="button">
        <svg class="icon-root lexicon-icon lexicon-icon-times">
          <use xlink:href="/images/lexicon/icons.svg#times"></use>
        </svg>
      </button>
    </div>
    <form class="form-root">
      <div class="form-group form-inline form-inline-autofit">
        <div class="">
          <label for="FormInput1" class="form-control-label label-root">new-field-name</label>
          <input placeholder="enter-new-field-name" id="FormInput1" value="" class="input-root form-control" type="text">
        </div>
        <div>
          <label>field-type</label>
          <div class="dropdown dropdown-root searchable-select-root" id="overlay3">
            <button class="button-root btn btn-secondary dropdown-toggle" type="button">
              <span class="text-truncate">select </span>
              <span>
                <svg class="icon-root lexicon-icon lexicon-icon-caret-double">
                  <use xlink:href="/images/lexicon/icons.svg#caret-double"></use>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div class="modal-footer">
        <button class="button-root btn btn-secondary" type="button">cancel</button>
        <button class="button-root btn btn-primary" type="submit">create</button>
      </div>
    </form>
  </div>
</div>
`;
